test/e2e-v2/script/docker-compose/base-compose.yml

```diff
@@ -0,0 +1,117 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+version: '2.1'
+
+services:
+  banyandb:
+    image: "skywalking-banyandb:${TAG}"
+    networks:
+      - e2e
+    expose:
+      - 17912
+    command: standalone --stream-root-path /tmp/stream-data --measure-root-path /tmp/measure-data
+    healthcheck:
+      test: ["CMD", "sh", "-c", "nc -nz 127.0.0.1 17912"]
+      interval: 5s
+      timeout: 60s
+      retries: 120
+
+  oap:
+    image: "ghcr.io/apache/skywalking/oap:${SW_OAP_COMMIT}"
+    expose:
+      - 11800
+      - 12800
+      - 10051
+      - 5005
+    networks:
+      - e2e
+    volumes:
+      - ./../prepare/setup-oap/log4j2.xml:/skywalking/config/log4j2.xml
+    environment:
+      SW_CLUSTER_ZK_HOST_PORT: zk:2181
+      SW_STORAGE_ES_CLUSTER_NODES: es:9200
+      SW_JDBC_URL: jdbc:mysql://mysql:3306/swtest
+      SW_STORAGE_INFLUXDB_URL: http://influxdb:8086
+      SW_STORAGE_BANYANDB_HOST: banyandb
+      SW_STORAGE_IOTDB_HOST: iotdb
+      SW_STORAGE_IOTDB_SESSIONPOOL_SIZE: 2
+      SW_CONFIG_ETCD_PERIOD: 1
+      SW_CONFIG_ETCD_ENDPOINTS: http://etcd:2379
+      SW_CLUSTER_ETCD_ENDPOINTS: http://etcd:2379
+    healthcheck:
+      test: ["CMD", "bash", "-c", "cat < /dev/null > /dev/tcp/127.0.0.1/11800"]
+      interval: 5s
+      timeout: 60s
+      retries: 120
+
+  ui:
+    image: "ghcr.io/apache/skywalking/ui:${SW_OAP_COMMIT}"
+    expose:
+      - 8080
+    networks:
+      - e2e
+    environment:
+      - SW_OAP_ADDRESS=http://oap:12800
+
+  provider:
+    image: "ghcr.io/apache/skywalking/e2e-service-provider:latest"
```

Comment:
Please lock the version in case of any breaks in the future(eventually it will), choose share-volume and make sure to use `SW_AGENT_JAVA_COMMIT` to get the locked agent version.